### DISCUSSION TITLE
Export in html the 3D grid from show_grid function

### DISCUSSION THREAD

**Discussion by qtn-mtp** [pyvista/pyvista#3775] [Q&A]
Hello,
I would like to export some 3D scenes in html and it works good enough for my simple geometric shapes. I am using the function [export_html](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_html.html#export-html) with the backend [pythreejs](https://docs.pyvista.org/user-guide/jupyter/pythreejs.html#using-pythreejs-with-pyvista).
However, the grid generated with the function [show_grid](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.show_grid.html#show-grid) is not exported in the html file.
For example, if I run the following code, my html file "./pv_sphere.html" contains the sphere but not the grid :
```python
import pyvista as pv
pl = pv.Plotter()
pl.set_background('white')
pl.show_grid(color='black')
sphere = pv.Sphere()
pl.add_mesh(sphere, color="blue", show_edges=True)
pl.export_html("./pv_sphere.html")
pl.show()
```
I understood it's because it's not supported as explained in the section [caveats from pythreejs](https://docs.pyvista.org/user-guide/jupyter/pythreejs.html#caveats)
So, would you know an alternative to have a 3D grid with label in html file ?

Another little "problem" (which is not related and not very important) is I have these following exceptions when I call [export_html](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_html.html#export-html) function even if the scene is empty :
```
Exception ignored in: <function BasePlotter.__del__ at 0x7ffa7389c0d0>
Traceback (most recent call last):
  File "/home/qmassone/anaconda3/lib/python3.8/site-packages/pyvista/plotting/plotting.py", line 5361, in __del__
  File "/home/qmassone/anaconda3/lib/python3.8/site-packages/pyvista/plotting/plotting.py", line 4084, in deep_clean
  File "/home/qmassone/anaconda3/lib/python3.8/site-packages/pyvista/plotting/picking.py", line 1280, in disable_picking
  File "/home/qmassone/anaconda3/lib/python3.8/site-packages/pyvista/plotting/plotting.py", line 1351, in remove_actor
  File "/home/qmassone/anaconda3/lib/python3.8/site-packages/pyvista/plotting/renderer.py", line 2155, in remove_actor
AttributeError: 'NoneType' object has no attribute 'abc'
```
```
qmassone@qmassone-OMEN-by-HP-Laptop:~$ conda list | grep "pyvista\|pythreejs\|ipywidgets"
ipywidgets                7.5.1                      py_0  
pythreejs                 2.4.1                    pypi_0    pypi
pyvista                   0.37.0                   pypi_0    pypi
```
Thanks in advance
Sincerly

**Comment by banesullivan** [comment]
The grid lines are not currently supported by the JavaScript rendering backends that the scene graph is converted to when exporting to HTML. This is a limitation of ThreeJS (or pythreejs) and mostly out of our control.

I am working to refactor/reqrite all of the Jupyter backend mechanisms and it's on my to-do list to refactor the HTML export to strictly use VTK.js. In which case, this should be possible but may require a bit of work to leverage the CubeAxes: https://kitware.github.io/vtk-js/examples/CubeAxes.html

**Reply by qtn-mtp** [reply]
Thanks for your answer. Indeed, I was reading pythreejs documentation and it's clearly not supported so yes it is out of your control.
So, in the meantime, you advice to use directly the library VTK.js ? But I imagine, it's not possible to mix this library with pyvista ?

**Reply by banesullivan** [reply]
It is possible! We currently have a very outdated `export_vtkjs()` method which I'm working to modernize - stand by and hopefully the next release will have this handled for you

**Comment by banesullivan** [comment]
> Another little "problem" (which is not related and not very important) is I have these following exceptions when I call [export_html](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.export_html.html#export-html) function even if the scene is empty :

Huh. well that doesn't look good. Is that the full traceback?

**Reply by qtn-mtp** [reply]
I think. I copied the output from my terminal in the case where I export an empty 3D scene (I have the same exceptions even if the scene is not empty)
